### PR TITLE
KEP-1669: update docs for ProxyTerminatingEndpoints

### PR DESCRIPTION
**Pull request by andrewsykim**
Signed-off-by: Andrew Sy Kim <andrewsy@google.com>

Update feature gate references for ProxyTerminatingEndpoints. 

https://github.com/kubernetes/enhancements/issues/1669

### CODE REVIEW

File: content/en/docs/concepts/services-networking/service.md [hunk 1]
```diff
@@ -518,7 +518,7 @@ and `Local` to only route to ready node-local endpoints. If the traffic policy i
 endpoints, the kube-proxy does not forward any traffic for the relevant Service.
 
 {{< note >}}
-{{< feature-state for_k8s_version="v1.22" state="alpha" >}}
+{{< feature-state for_k8s_version="v1.26" state="beta" >}}
 If you enable the `ProxyTerminatingEndpoints`
 [feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
 for the kube-proxy, the kube-proxy checks if the node
```
Comment (sftim) [inline content/en/docs/concepts/services-networking/service.md:524]:
```suggestion
If the `ProxyTerminatingEndpoints`
[feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
is enabled for the kube-proxy on a node, that node's kube-proxy uses a more
complicated algorithm to select endpoints for a Service. With the feature
enabled, the kube-proxy checks if the node
```

Comment (andrewsykim) [reply content/en/docs/concepts/services-networking/service.md]: Thanks, used some of your suggestion and made some addiional changes

File: content/en/docs/concepts/services-networking/service.md [hunk 2]
```diff
@@ -542,6 +526,28 @@ Valid values are `Cluster` and `Local`. Set the field to `Cluster` to route inte
...
+has local endpoints and whether or not all the local endpoints are marked as terminating.
+If there are local endpoints and **all** of them are terminating, then kube-proxy
+will forward traffic to those terminating endpoints. Otherwise, kube-proxy will always
+prefer forwarding traffic to endpoints that are not terminating.
+
+This forwarding behavior for terminating endpoints exist to allow `NodePort` and `LoadBalancer` Services to
+gracefully drain connections when using `externalTrafficPolicy=Local`. As a deployment goes through
+a rolling update, nodes backing a loadbalancer may transition from N to 0 replicas of that deployment.
+In some cases, external load balancers can send traffic to a node with 0 replicas in between health check probes.
+Routing traffic to termiating endpoints ensures that Node's that are scaling down Pods can gracefully receive
```
Comment (mengjiao-liu) [inline content/en/docs/concepts/services-networking/service.md:547]:
```suggestion
Routing traffic to terminating endpoints ensures that Node's that are scaling down Pods can gracefully receive
```

**Comment by krol3** [comment]
Hi @andrewsykim! This PR needs a doc review by **Mon Nov 28th** to get this into the release. Please reach out to required SIGs to get their review. Thank you!

**Comment by krol3** [comment]
@andrewsykim  please can you review the suggesting changes?
This PR needs a doc review by **Mon Nov 28th** to get this into the release. Please reach out to required SIGs to get their review. Thank you!

**Comment by andrewsykim** [comment]
> @andrewsykim please can you review the suggesting changes?
This PR needs a doc review by Mon Nov 28th to get this into the release. Please reach out to required SIGs to get their review. Thank you!

done, thanks

**Review by krol3 — APPROVED** [review]
/lgtm

**Pull request merged**